x-pack/filebeat/input/httpjson/config.go

```diff
@@ -65,6 +67,43 @@ type RateLimit struct {
 	Remaining string `config:"remaining"`
 }
 
+type DateCursor struct {
+	Enabled         *bool         `config:"enabled"`
+	Field           string        `config:"field" validate:"required"`
+	URLField        string        `config:"url_field" validate:"required"`
+	ValueTemplate   *Template     `config:"value_template"`
+	DateFormat      string        `config:"date_format"`
+	InitialInterval time.Duration `config:"initial_interval"`
+}
+
+type Template struct {
+	*template.Template
+}
+
+func (t *Template) Unpack(in string) error {
+	tpl, err := template.New("tpl").Parse(in)
+	if err != nil {
+		return err
+	}
+
+	*t = Template{Template: tpl}
+
+	return nil
+}
+
+// IsEnabled returns true if the `enable` field is set to true in the yaml.
+func (dc *DateCursor) IsEnabled() bool {
+	return dc != nil && (dc.Enabled == nil || *dc.Enabled)
+}
+
+// IsEnabled returns true if the `enable` field is set to true in the yaml.
+func (dc *DateCursor) GetDateFormat() string {
+	if dc.DateFormat == "" {
+		return time.RFC3339
+	}
+	return dc.DateFormat
+}
+
 func (c *config) Validate() error {
```

Comment:
Should we add any validation for any of the DateCursor items? For example dateformat that it's a valid format maybe?

Reply:
Another validation might be to prevent this:
`date_cursor field must be a string, cursor will not advance	{"url": "https://api.securitycenter.windows.com/api/alerts"}`

Reply:
Not sure if we can add a validation for this since at validation time we still do not have any body to check if the path to the object is correct. Maybe a better error message if not found would be good though, instead of telling it is not a string.

Reply:
It should be fine for now. It seems that it only happens if I created a config entry with dot notation without quotes around it, so it was not treating it as a string object.